File: packages/alteration/src/main/java/fdit/alteration/core/incident/IncidentDeserializer.java
package fdit.alteration.core.incident;

import com.fasterxml.jackson.databind.ObjectMapper;
import fdit.alteration.core.utils.StreamUtils;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;

public class IncidentDeserializer {

    private final File incidentFile;

    public IncidentDeserializer(final File incidentFile) {
        this.incidentFile = incidentFile;
    }

    public Incident deserialize(ObjectMapper mapper) throws IOException {
        final String content = StreamUtils.inputStreamToString(Files.newInputStream(incidentFile.toPath()));
        return mapper.readValue(content, Incident.class);
    }
}

File: packages/fdit-ts/src/Incident/IncidentDeserializer.ts
import * as fs from 'fs'
import { Incident } from './Incident'

export class IncidentDeserializer {
  private incidentFile: string

  constructor(incidentFile: string) {
    this.incidentFile = incidentFile
  }

  public deserialize(): Incident {
    const content = fs.readFileSync(this.incidentFile, 'utf8')
    return JSON.parse(content) as Incident
  }
}
